pkg/gofr/datasource/redis/redis_test.go

```diff
@@ -0,0 +1,115 @@
+package redis
+
+import (
+	"context"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/alicebob/miniredis/v2"
+	"github.com/go-redis/redismock/v9"
+	"github.com/stretchr/testify/assert"
+
+	"go.uber.org/mock/gomock"
+
+	"gofr.dev/pkg/gofr/datasource"
+	"gofr.dev/pkg/gofr/testutil"
+)
+
+func TestRedis_QueryLogging(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	// Mock Redis server setup
+	s, err := miniredis.Run()
+	assert.Nil(t, err)
+
+	defer s.Close()
+
+	// Convert port to integer
+	port, err := strconv.Atoi(s.Port())
+	assert.Nil(t, err)
+
+	// Mock Redis client setup
+	_, mockClient := redismock.NewClientMock()
+	mockClient.ExpectPing().SetVal("PONG")
+	mockClient.ExpectSet("key", "value", 1*time.Minute).SetVal("OK")
```

Comment:
Add the reason behind client and server setup - will be good in future references

Reply:
We don't need both of them will correct the code